### ISSUE TITLE
Checkin with globus is responding with a 500 error

### ISSUE THREAD

**Issue by JBorrow** [simonsobs/librarian#95]
Need to remedy this to re-enable Princeton transfers.

**Comment by iparask** [comment]
Can you paste an error message here? Thank you!

**Comment by JBorrow** [comment]
```(requests:request) [2024-11-11 18:57:18,442] {DEBUG}:transport request retry cycle. attempt=5
(requests:request) [2024-11-11 18:57:18,442] {DEBUG}:request about to send
(connectionpool:_make_request) [2024-11-11 18:57:18,562] {DEBUG}:https://auth.globus.org:443 "POST /v2/oauth2/token HTTP/11" 500 246
(requests:request) [2024-11-11 18:57:18,562] {DEBUG}:request success, still check should-retry
(retry:should_retry) [2024-11-11 18:57:18,562] {DEBUG}:ran retry check (default_check_request_exception) => RetryCheckResult.no_decision
(retry:should_retry) [2024-11-11 18:57:18,562] {DEBUG}:ran retry check (default_check_retry_after_header) => RetryCheckResult.no_decision
(retry:should_retry) [2024-11-11 18:57:18,562] {DEBUG}:ran retry check (default_check_transient_error) => RetryCheckResult.do_retry
(requests:request) [2024-11-11 18:57:18,562] {DEBUG}:request may retry, will check attempts
(requests:request) [2024-11-11 18:57:18,562] {WARNING}:request reached max retries, done (fail, response)
(client:request) [2024-11-11 18:57:18,562] {DEBUG}:request made to URL: https://auth.globus.org/v2/oauth2/token
(client:request) [2024-11-11 18:57:18,562] {DEBUG}:request completed with (error) response code: 500
(core:_run_job) [2024-11-11 18:57:18,566] {ERROR}:Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/librarian_background/core.py", line 38, in _run_job
    super()._run_job(job)
  File "/usr/local/lib/python3.11/site-packages/schedule/__init__.py", line 173, in _run_job
    ret = job.run()
          ^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/schedule/__init__.py", line 691, in run
    ret = self.job_func()
          ^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/librarian_background/task.py", line 41, in __call__
    return self.on_call()
           ^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/librarian_background/queues.py", line 85, in on_call
    self.core(session_maker=get_session)
  File "/usr/local/lib/python3.11/site-packages/librarian_background/queues.py", line 76, in core
    check_on_consumed(
  File "/usr/local/lib/python3.11/site-packages/librarian_background/queues.py", line 137, in check_on_consumed
    current_status = queue_item.async_transfer_manager.transfer_status(
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/hera_librarian/async_transfers/globus.py", line 246, in transfer_status
    task_doc = transfer_client.get_task(self.task_id)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/globus_sdk/services/transfer/client.py", line 1614, in get_task
    return self.get(f"task/{task_id}", query_params=query_params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/globus_sdk/client.py", line 139, in get
    return self.request("GET", path, query_params=query_params, headers=headers)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/globus_sdk/client.py", line 278, in request
    r = self.transport.request(
        ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/globus_sdk/transport/requests.py", line 310, in request
    self._set_authz_header(authorizer, req)
  File "/usr/local/lib/python3.11/site-packages/globus_sdk/transport/requests.py", line 246, in _set_authz_header
    authz_header = authorizer.get_authorization_header()
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/globus_sdk/authorizers/renewing.py", line 168, in get_authorization_header
    self.ensure_valid_token()
  File "/usr/local/lib/python3.11/site-packages/globus_sdk/authorizers/renewing.py", line 162, in ensure_valid_token
    self._get_new_access_token()
  File "/usr/local/lib/python3.11/site-packages/globus_sdk/authorizers/renewing.py", line 122, in _get_new_access_token
    res = self._get_token_response()
          ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/globus_sdk/authorizers/refresh_token.py", line 85, in _get_token_response
    return self.auth_client.oauth2_refresh_token(self.refresh_token)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/globus_sdk/services/auth/client/native_client.py", line 134, in oauth2_refresh_token
    return self.oauth2_token(form_data, body_params=body_params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/globus_sdk/services/auth/client/base_login_client.py", line 396, in oauth2_token
    self.post(
  File "/usr/local/lib/python3.11/site-packages/globus_sdk/client.py", line 156, in post
    return self.request(
           ^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/globus_sdk/client.py", line 296, in request
    raise self.error_class(r)
globus_sdk.services.auth.errors.AuthAPIError: ('POST', 'https://auth.globus.org/v2/oauth2/token', None, 500, 'UNEXPECTED_ERROR', 'An unexpected error occurred.', '415d9b9d-dace-4037-8113-3cf4cb2b653d')
```

**Comment by iparask** [comment]
Do you happen to have the job description when this is happening? I remember you said it was when librarian tries to verify the transfer state. Is that correct?

**Comment by JBorrow** [comment]
Yes this occurs here: https://github.com/simonsobs/librarian/blob/8ce3ab23bae1dfcb120a77384942630909776393/librarian_background/queues.py#L137C15-L137C17

**Comment by iparask** [comment]
I cannot replicate it with a script that does the following:
```python
import globus_sdk

CLIENT_ID = "..."
REFRESH_TOKEN = "..."
PULIBRARIAN_ENDPOINT = "..."
NERSC_ENDPOINT = "..."

if __name__ == "__main__":

    client = globus_sdk.NativeAppAuthClient(client_id=CLIENT_ID)
    authorizer = globus_sdk.RefreshTokenAuthorizer(refresh_token=REFRESH_TOKEN, auth_client=client)
    task_id = "e9002926-9e41-11ef-8670-73df88a31e54"
    transfer_client = globus_sdk.TransferClient(authorizer=authorizer)
    task_doc = transfer_client.get_task(task_id)
    print(task_doc)
```

I run this from the librarian pod on NERSC and the librarian VM at Princeton. The response is as expected (I emitted sensitive info) :
```json
{
  "DATA_TYPE": "task",
  "bytes_checksummed": 0,
  "bytes_transferred": 44196014674,
  "canceled_by_admin": null,
  "canceled_by_admin_message": null,
  "completion_time": "2024-11-09T02:34:28+00:00",
  "deadline": "2024-11-10T02:25:40+00:00",
  "destination_endpoint_display_name": "pulibrarian",
  "destination_endpoint_id": "...",
  "destination_local_user": null,
  "destination_local_user_status": null,
  "directories": 1258,
  "effective_bytes_per_second": 83717237,
  "encrypt_data": false,
  "fail_on_quota_errors": false,
  "fatal_error": null,
  "faults": 0,
  "files": 1211,
  "files_skipped": 0,
  "files_transferred": 1211,
  "filter_rules": null,
  "history_deleted": false,
  "is_ok": null,
  "is_paused": false,
  "label": "batch with oper_1684683543_lati6_100",
  "nice_status": null,
  "nice_status_details": null,
  "nice_status_expires_in": null,
  "nice_status_short_description": null,
  "preserve_timestamp": true,
  "recursive_symlinks": "ignore",
  "request_time": "2024-11-09T02:25:40+00:00",
  "skip_source_errors": false,
  "source_endpoint_display_name": "NERSC DTN sobs Collab",
  "source_local_user": null,
  "source_local_user_status": null,
  "status": "SUCCEEDED",
  "subtasks_canceled": 0,
  "subtasks_expired": 0,
  "subtasks_failed": 0,
  "subtasks_pending": 0,
  "subtasks_retrying": 0,
  "subtasks_skipped_errors": 0,
  "subtasks_succeeded": 2525,
  "subtasks_total": 2525,
  "symlinks": 0,
  "sync_level": 0,
  "task_id": "e9002926-9e41-11ef-8670-73df88a31e54",
  "type": "TRANSFER",
  "verify_checksum": true
}

**Comment by JBorrow** [comment]
Could be because we are not using the merged version of #78

**Comment by JBorrow** [comment]
Using #78 does indeed resolve this problem. We get this 500 because we are using old creds to try to access a new item. That's a problem (should be 400-level error) but not our problem. 500 reported to Globus.
